### PR TITLE
Use a secret for storing osephemeralheat password

### PR DESCRIPTION
**Pull request by dprince**

### CODE REVIEW

File: controllers/openstackephemeralheat_controller.go [hunk 1]
```diff
@@ -96,13 +96,25 @@ func (r *OpenStackEphemeralHeatReconciler) Reconcile(ctx context.Context, req ct
...
+			r.Log.Info(fmt.Sprintf("Secret %s successfully reconciled - operation: ephemeral-heat-%s", instance.Name, string(op)))
+		}
+	}
+
 	envVars := make(map[string]common.EnvSetter)
 	templateParameters := make(map[string]interface{})
 	templateParameters["MariaDBHost"] = "mariadb-" + instance.Name
 	templateParameters["RabbitMQHost"] = "rabbitmq-" + instance.Name
-	templateParameters["MariaDBPassword"] = password
+	templateParameters["MariaDBPassword"] = passwordSecret.Data["password"]
```
Comment (dprince) [inline controllers/openstackephemeralheat_controller.go:117]: This isn't quite right yet. Need to either convert to string and/or base64.decode?

Comment (abays) [reply controllers/openstackephemeralheat_controller.go:117]: I think a base64 decode might do the trick

Comment (dprince) [reply controllers/openstackephemeralheat_controller.go]: so I tried just adding string() and it seems to work fine. You do need to base64.decode if you access it from the CLI but apparently not via the Golang libraries

**Comment by dprince** [comment]
Tested and this seems to functionally work fine.

**Review by stuggi — APPROVED** [review]
/lgtm

**Pull request merged**